src/Compilers/Core/Portable/WellKnownMembers.cs

```diff
@@ -3526,6 +3526,15 @@ static WellKnownMembers()
                 0,                                                                                                          // Arity
                     0,                                                                                                      // Method Signature
                     (byte)SignatureTypeCode.TypeHandle, (byte)SpecialType.System_String, // Return Type
+
+                // System_Index__op_Implicit_FromInt32
```

Comment:
@333fred raised a good question in Teams, whether the spec calls out this conversion operator. The lowering section in the spec has two examples, which imply different lowering strategies. One implies use of conversion (`expr[0]`) and the other explicitly uses construction (`expr[new Index(0)]`).
I wouldn't mind aligning on constructor (since that's what we use for "from end" Index values) and updating the spec accordingly. What do you think?

Reply:
Could you provide a link to the relevant section?

Reply:
Lowering section: https://github.com/dotnet/csharplang/blob/main/proposals/list-patterns.md#lowering

Reply:
> I wouldn't mind aligning on constructor (since that's what we use for "from end" Index values) and updating the spec accordingly. What do you think?

Either way would be fine, I think. However, aligning on constructor requires changes in DetermineMakePatternIndexOffsetExpressionStrategy

Reply:
Added a note to [https://github.com/dotnet/roslyn/issues/57825](https://github.com/dotnet/roslyn/issues/57825) (follow-up issue for conversion from int to Index)